### PR TITLE
adding support for CONFIG GET

### PR DESCRIPTION
**Pull request by ipapapa**
Adding support for CONFIG request operations that are not changing the Redis configuration.

### CODE REVIEW

File: src/dyn_message.h [hunk 1]
```diff
@@ -183,6 +183,7 @@ typedef enum msg_type {
     MSG_RSP_REDIS_INTEGER,
     MSG_RSP_REDIS_BULK,
     MSG_RSP_REDIS_MULTIBULK,
+	MSG_REQ_REDIS_CONFIG,
```
Comment (shailesh33) [inline src/dyn_message.h]: I think this should be MSG_REQ_REDIS_CONFIG_GET. that way it is more specific.

File: src/proto/dyn_redis.c [hunk 2]
```diff
@@ -1108,6 +1118,8 @@ redis_parse_req(struct msg *r)
                     break;
                 }
 
+                break;
```
Comment (shailesh33) [inline src/proto/dyn_redis.c]: good catch.

File: notes/redis.md [hunk 3]
```diff
@@ -341,11 +341,13 @@
...
-    |    CONFIG GET     |    No      | CONFIG GET parameter                                                                                                |
+    |    CONFIG GET     |    YES     | CONFIG GET parameter                                                                                                |
     +-------------------+------------+---------------------------------------------------------------------------------------------------------------------+
     |    CONFIG SET     |    No      | CONFIG SET parameter value                                                                                          |
     +-------------------+------------+---------------------------------------------------------------------------------------------------------------------+
-    |  CONFIG RESETSTAT |    No      | CONFIG RESETSTAT                                                                                                    |
+    |  CONFIG RESETSTAT |    YES     | CONFIG RESETSTAT                                                                                                    |
```
Comment (shailesh33) [inline notes/redis.md]: where is this implemented?

File: src/dyn_message.h (hunk 1 above)
Comment (ipapapa) [inline src/dyn_message.h]: I disagree with the this. When we do the string comparison we look for "CONFIG". Later we we remove all others, other than "GET".

File: notes/redis.md (hunk 3 above)
Comment (ipapapa) [inline notes/redis.md]: fixed it, sorry

File: src/proto/dyn_redis.c [hunk 4]
```diff
@@ -1393,7 +1406,14 @@ redis_parse_req(struct msg *r)
             break;
 
         case SW_ARG1:
+
+
+            if (r->type == MSG_REQ_REDIS_CONFIG && !str3icmp(m, 'g', 'e', 't')) {
```
Comment (shailesh33) [inline src/proto/dyn_redis.c]: shouldnt it be r->type == MSG_REQ_REDIS_CONFIG && str3icmp(m, 'g', 'e', 't')

you want to error out for anything other than get, and for that str3icmp would return non 0,

**Pull request merged**